addrmgr/addrmanager_test.go

```diff
@@ -558,3 +678,328 @@ func TestCorruptPeersFile(t *testing.T) {
+	const routableIpv6Address = "2003::"
+	onionCatTorV2Address := onionCatNet.IP.String()
+	rfc4380IPAddress := rfc4380Net.IP.String()
+	rfc3964IPAddress := rfc3964Net.IP.String()
+	rfc6052IPAddress := rfc6052Net.IP.String()
+	rfc6145IPAddress := rfc6145Net.IP.String()
+
+	tests := []struct {
+		name          string
+		localAddress  string
+		remoteAddress string
+		valid         bool
+		reach         int
+	}{
+		{
+			name:          "torv2 to torv2",
+			localAddress:  onionCatTorV2Address,
+			remoteAddress: onionCatTorV2Address,
+			valid:         false,
+			reach:         Private,
+		},
+		{
+			name:          "routable ipv4 to torv2",
+			localAddress:  routableIpv4Address,
+			remoteAddress: onionCatTorV2Address,
+			valid:         true,
+			reach:         Ipv4,
+		},
+		{
+			name:          "not routable ipv4 to torv2",
+			localAddress:  unroutableIpv4Address,
+			remoteAddress: onionCatTorV2Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "routable ipv6 to torv2",
+			localAddress:  routableIpv6Address,
+			remoteAddress: onionCatTorV2Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "not routable ipv6 to torv2",
+			localAddress:  unroutableIpv6Address,
+			remoteAddress: onionCatTorV2Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "rfc4380 to rfc4380",
+			localAddress:  rfc4380IPAddress,
+			remoteAddress: rfc4380IPAddress,
+			valid:         true,
+			reach:         Teredo,
+		},
+		{
+			name:          "not routable ipv4 to rfc4380",
+			localAddress:  unroutableIpv4Address,
+			remoteAddress: rfc4380IPAddress,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "routable ipv4 to rfc4380",
+			localAddress:  routableIpv4Address,
+			remoteAddress: rfc4380IPAddress,
+			valid:         true,
+			reach:         Ipv4,
+		},
+		{
+			name:          "routable ipv6 to rfc4380",
+			localAddress:  routableIpv6Address,
+			remoteAddress: rfc4380IPAddress,
+			valid:         true,
+			reach:         Ipv6Weak,
+		},
+		{
+			name:          "routable ipv4 to routable ipv4",
+			localAddress:  routableIpv4Address,
+			remoteAddress: routableIpv4Address,
+			valid:         true,
+			reach:         Ipv4,
+		},
+		{
+			name:          "routable ipv6 to routable ipv4",
+			localAddress:  routableIpv6Address,
+			remoteAddress: routableIpv4Address,
+			valid:         false,
+			reach:         Unreachable,
+		},
+		{
+			name:          "not routable ipv4 to routable ipv6",
+			localAddress:  unroutableIpv4Address,
+			remoteAddress: routableIpv6Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "not routable ipv6 to routable ipv6",
+			localAddress:  unroutableIpv6Address,
+			remoteAddress: routableIpv6Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "not routable to routable",
```

Comment:
s / "not routable to routable" / "unroutable ipv4 to routabe ipv6"

Reply:
Updated